lib/chai/core/assertions.js

```diff
@@ -537,8 +537,8 @@ module.exports = function (chai, _) {
         itemsCount = val.length;
         break;
       case 'function':
-        var name = val.name ? ' ' + val.name : '';
-        throw new TypeError('.empty was passed a function' + name);
+        var msg = '.empty was passed a function ' + _.getName(val);
+        throw new TypeError(msg.trim());
```

Comment:
@meeber exactly. Great explanation :+1: 
And, as you said in the last comment, if someone explicitly added spaces to a function's name, I think we should keep it that way.